sympy/integrals/tests/test_manual.py

```diff
@@ -70,6 +70,12 @@ def test_manualintegrate_trigonometry():
 
     assert manualintegrate(sec(x), x) == log(sec(x) + tan(x))
     assert manualintegrate(csc(x), x) == -log(csc(x) + cot(x))
+    #Test for Issue#12558
+    assert manualintegrate(sec(4*x)**2, x) == tan(4*x)/4
+    assert manualintegrate(csc(2*x)**2, x) == -cot(2*x)/2
+    assert manualintegrate(sec(2*x), x) == log(tan(2*x) + sec(2*x)) / 2
+    assert manualintegrate(csc(4*x), x) == -log(cot(4*x) + csc(4*x)) / 4
+    assert manualintegrate(sec(x**2)**2, x) == (tan(x**2)) / (2*x)
```

Comment:
This is still not correct

Reply:
Can you please explain it to me,  I don't understand why it is not correct?

Reply:
Derivative of `tan(x**2)/(2(x)` is not `sec(x**2)**2`

Reply:
So the integration of sec(x**2)**2 is not possible.